### DISCUSSION TITLE
It seems the App source '...' is not there

### DISCUSSION THREAD

**Discussion by alexreg** [Homebrew/discussions#656] [Casks]
I am trying to create a cask formula for [GDAT](https://www.getgmp.com/), and it's not finding the `.app` file after extracting the ZIP. I tried to look at the caskroom dir to which it extracted things, it not longer exists by the time the command finishes, evidently. Even if I specify `--keep-tmp`, that doesn't help.

```
==> Downloading https://drive.google.com/file/d/1lnA5hFiCV9qzreQE7a5YzdeValaicTHD/view?usp=sharing
Already downloaded: /Library/Caches/Homebrew/downloads/dc2452c94761e88d0f06f0739849ed134a00363a7de1cc5ffb705ab56d4ff63b--view?usp=sharing
==> Installing Cask gdat
==> Purging files for version 2021r01 of Cask gdat
Error: It seems the App source '/usr/local/Caskroom/gdat/2021r01/OS X 64 bit/Genealogical DNA Analysis Tool.app' is not there.
```

Any ideas?

**Comment by gromgit** [comment]
Can you show us your cask definition?

**Reply by alexreg** [reply]
```ruby
cask "gdat" do
  version "2021r01"
  sha256 "6b625131e0d71013bbaef51cd71291bd4f992226e98e24ccc4f22cfe1261b745"

  url "https://drive.google.com/file/d/1lnA5hFiCV9qzreQE7a5YzdeValaicTHD/view?usp=sharing"
  name "Genealogical DNA Analysis Tool"
  desc "An app that utilizes autosomal DNA to aid in the research of family trees"
  homepage "https://sites.google.com/view/genealogical-dna-analysis-tool"

  livecheck do
    url "https://sites.google.com/view/genealogical-dna-analysis-tool/home/download"
    strategy :page_match
    regex(/Genealogical DNA Analysis Tool \d{4}r\d{2}/i)
  end

  app "OS X 64 bit/Genealogical DNA Analysis Tool.app"

  depends_on arch: :intel
end
```

**Reply by gromgit** [reply]
I'm not sure how you got the error you did, but there are two problems with your cask definition:
* Your `url` stanza points to the Google Drive download page, which causes Homebrew to download HTML rather than the expected ZIP.
* Even after I manually downloaded the ZIP and replaced the abovementioned cache file, your `sha256` stanza doesn't match the ZIP checksum.

I can only assume that you actually have a different (older?) ZIP in your cache, so Homebrew didn't attempt to download it again. That's the only reason I can think of for both the `url` and `sha256` discrepancies to go undetected.

For the record, the following are the correct values for both, as far as I can tell:
* `url "https://drive.google.com/uc?export=download&id=1lnA5hFiCV9qzreQE7a5YzdeValaicTHD"`
* `sha256 "23046246d101206b911298d78b751b036c8acb44eb4420dba62ce0a332405399"`

I was able to install GDAT without any errors, after making the above fixes.

**Reply by alexreg** [reply]
Thanks, works for me too! Should have realised.
